packages/web/src/components/common/reward-tooltip-content/RewardTooltipContent.tsx

```diff
@@ -22,140 +23,57 @@ export interface RewardTooltipContentProps {
-    });
-  }, [swapFeeRewards]);
-
-  const internalRewardUSD = useMemo(() => {
-    const isEmpty = !internalRewards;
-
-    if (isEmpty) return "-";
-
-    const sumUSD = internalRewards.reduce((accum: null | number, current) => {
-      if (accum === null && current.usd === null) {
-        return null;
-      }
-
-      if (accum === null) {
-        return current.usd;
-      }
-
-      if (current.usd === null) {
-        return accum;
-      }
-
-      return accum + current.usd;
-    }, null);
-    return formatOtherPrice(sumUSD, {
-      isKMB: false,
-    });
-  }, [internalRewards]);
-
-  const externalRewardUSD = useMemo(() => {
-    const isEmpty = !externalRewards;
-
-    if (isEmpty) return "-";
-
-    const sumUSD = externalRewards.reduce((accum: null | number, current) => {
-      if (accum === null && current.usd === null) {
-        return null;
-      }
-
-      if (accum === null) {
-        return current.usd;
-      }
-
-      if (current.usd === null) {
-        return accum;
-      }
-
-      return accum + current.usd;
-    }, null);
-    return formatOtherPrice(sumUSD, {
-      isKMB: false,
-    });
-  }, [externalRewards]);
+  const sortByUsd = React.useCallback(
+    (rewards: PositionRewardForTooltip[]) => [...rewards].sort((a, b) => (b.usd ?? 0) - (a.usd ?? 0)),
+    [],
+  );
 
-  const noneRewardUSD = useMemo(() => {
-    const isEmpty = !noneRewards || noneRewards?.length === 0;
+  const calculateTotalUsd = React.useCallback((rewards: PositionRewardForTooltip[] | null): string => {
+    if (!rewards || rewards.length === 0) return "-";
 
-    if (isEmpty) return "-";
+    const sumUSD = rewards.reduce(
+      (accum, current) => (current.usd !== null ? accum.plus(current.usd) : accum),
+      new BigNumber(0),
+    );
 
-    const sumUSD = noneRewards.reduce((accum: null | number, current) => {
-      if (accum === null && current.usd === null) {
-        return null;
-      }
+    return formatOtherPrice(sumUSD.toNumber(), { isKMB: false });
+  }, []);
 
-      if (accum === null) {
-        return current.usd;
+  const processRewardsByType = React.useCallback(
+    (type: DisplayRewardType): { rewards: PositionRewardForTooltip[] | null; totalUSD: string } => {
+      if (!rewardInfo || rewardInfo[type].length === 0) {
+        return { rewards: null, totalUSD: "-" };
       }
 
-      if (current.usd === null) {
-        return accum;
-      }
+      const sortedRewards = sortByUsd([...rewardInfo[type]]);
+      return {
+        rewards: sortedRewards,
+        totalUSD: calculateTotalUsd(sortedRewards),
+      };
+    },
+    [rewardInfo, sortByUsd, calculateTotalUsd],
+  );
 
-      return accum + current.usd;
-    }, null);
-    return formatOtherPrice(sumUSD, {
-      isKMB: false,
-    });
-  }, [noneRewards]);
+  const rewardsData = useMemo<RewardCategory[]>(() => {
+    const types: DisplayRewardType[] = ["SWAP_FEE", "INTERNAL_REWARD", "EXTERNAL_REWARD", "NONE"]; // Specify sort order
```

Comment:
Currently, the values of `DisplayRewardType` are hardcoded. Since `DisplayRewardType` is defined as a union type, if you're using TypeScript's type system, you could automatically extract all possible values.

Would you consider this approach if possible? Implementing this would ensure that when the type changes, the related constants are automatically updated, preventing mismatches between types and values in advance.

No need to apply this right away.